statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,245 @@
+import numpy as np 
+import pandas as pd
+#from numbers import Number
+
+
+class SurveyStat(object):
+
+    def __init__(self, data, cluster=None, strata=None, prob_weights=None):
+
+        data, self.cluster = self._check_type(data, cluster)
+        data, self.strata = self._check_type(data, strata)
+        data, self.prob_weights = self._check_type(data, prob_weights)
+        
+        n = data.shape[0]
+        p = data.shape[1]
+        self.data = np.asarray(data).reshape(n,p)
+
+    # preprocessing to be used by init
+    def _check_type(self, data, sampling_method):
+        """
+        converts cluster, strata, etc to an numpy array
+        gets rid of column from data if possible
+
+        Parameters
+        ----------
+        data : array-like
+          The raw data
+        sampling_method : one of cluster, strata, or prob_weight
+
+        Returns
+        -------
+        A nx1 numpy array
+        """
+
+        # check if str, num, or list
+        if isinstance(sampling_method, str):
```

Comment:
I'm not sure we should try to support integer column positions.  Column numbers in dataframes tend to change as the dataframe is manipulated so it's not a good practice from the user side.  I think we should support two inputs:

* If cluster, etc. are strings, you must be able to use `data[cluster]` to extract the column, similarly for weight and strata.  Then you convert the remainder to a ndarray and treat it as the data.

* Otherwise, cluster, strings, etc. must be array-like and can be converted to 1d ndarrays using np.asarray.  In this case, all of data is analyzed (nothing is dropped).

Don't worry too much about removing cluster, etc. (in the first case above) from data generically.  We can work out the details on this later.